src/Application/GameStates/GameFSMBuilder.cpp

```diff
@@ -0,0 +1,46 @@
+#include "GameFSMBuilder.h"
+
+#include <Engine/Engine.h>
+#include <Library/Fsm/FSMBuilder.h>
+
+#include <utility>
+#include <memory>
+
+#include "VideoState.h"
+
+std::unique_ptr<FSM> GameFSMBuilder::buildFSM() {
+    FSMBuilder fsmBuilder;
+    _buildIntroVideoSequence(fsmBuilder);
+
+    auto fsm = fsmBuilder.build();
+    _setStartingState(*fsm);
+    return std::move(fsm);
+}
+
+void GameFSMBuilder::_setStartingState(FSM &fsm) {
+    GameConfig::Debug &debugConfig = engine->config->debug;
+    if (!debugConfig.NoVideo.value()) {
+        if (!debugConfig.NoLogo.value()) {
+            fsm.jumpToState("3DOVideo");
+        } else if (!debugConfig.NoIntro.value()) {
+            fsm.jumpToState("IntroVideo");
+        }
+    }
```

Comment:
Yes, that makes sense. I can create a "StartupState" or something similar that performs a passthrough transition based on those config values. And it will use the same state paradigm.